### PR TITLE
S14672 Search "in:[username]" returns results in DMs

### PR DESCRIPTION
**Pull request by composednitin**
#### Summary

Added e2e test to validate search result in search box with query "in:[username] search-word" 

#### Ticket Link
Fixes  https://github.com/mattermost/mattermost-server/issues/10481

### CODE REVIEW

File: cypress/integration/search/search_user_post.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        cy.createNewDMChannel(users['user-2'].email);
+
+        // # Post message to user
+        cy.postMessage(message + '{enter}');
+
+        // # Type `in:` in searchbox
+        cy.get('#searchBox').type('in:');
+
+        // # Select user from suggestion list
+        cy.get('#search-autocomplete__popover').within(() => {
```
Comment (thekiiingbob) [inline cypress/integration/search/search_user_post.js]: This failed for me based on the existing data I had, I think we probably want to target the item that includes `@username` vs just `username`.

```suggestion
        cy.contains('.search-autocomplete__item', `@${users['user-2'].username}`).click();
```

File: cypress/integration/search/search_user_post.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        });
+
+        // # Validate searchbox contains the username
+        cy.get('#searchBox').should('have.value', 'in:@' + users['user-2'].username + ' ');
+
+        // # Press Enter in searchbox
+        cy.get('#searchBox').type(message).type('{enter}');
+
+        // # Search message in each filtered result
+        cy.get('#search-items-container').find('.search-item__container').each(($el) => {
```
Comment (thekiiingbob) [inline cypress/integration/search/search_user_post.js]: Just get the search highlight earlier like this:

```suggestion
        cy.get('#search-items-container').find('.search-highlight').each(($el) => {
            cy.wrap($el).should('have.text', message);
        });
```

File: cypress/support/ui_commands.js [hunk 3]
```diff
@@ -388,3 +388,17 @@ Cypress.Commands.add('updateChannelHeader', (text) => {
...
 });
+
+/**
+* create new DM channel
+* @param {String} text - DM channel name
+*/
+Cypress.Commands.add('createNewDMChannel', (channelname) => {
+    cy.get('h4#directChannel').scrollIntoView().find('button').should('contain', '+').click();
+
+    cy.get('#selectItems').within(() => {
```
[the remark this answers is not in the document]
Comment (thekiiingbob) [reply cypress/support/ui_commands.js]: Tried it out, force is required. Weird.

**Comment by composednitin** [comment]
@saturninoabril @thekiiingbob could you please review it again

File: cypress/integration/search/search_user_post.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+ * @param {String} text - DM channel name
+ */
+function createNewDMChannel(channelname) {
+    cy.get('#addDirectChannel').scrollIntoView().click();
+
+    cy.get('#selectItems').within(() => {
+        cy.get('input[type="text"]').scrollIntoView().type(channelname, {force: true});
+    });
+
+    // cy.get('div .clickable').children().contains(channelname).click({force: true});
```
Comment (saturninoabril) [inline cypress/integration/search/search_user_post.js]: Probably forgot to remove?

File: cypress/integration/search/search_user_post.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    cy.contains('.more-modal__description', channelname).click({force: true});
+    cy.get('#saveItems').click();
+}
+
+describe('Search in DMs', () => {
+    it('S14672 Search "in:[username]" returns results in DMs', () => {
+        // # Login and navigate to the app
+        cy.apiLogin('user-1');
+        cy.visit('/');
+        const message = 'Hello';
```
Comment (saturninoabril) [inline cypress/integration/search/search_user_post.js]: Can we make this a random message, something like?
```suggestion
        const message = 'Hello' + Date.now();
```

**Comment by composednitin** [comment]
@saturninoabril could you please review and merge

**Pull request merged**